app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -50,14 +51,51 @@ class CrimeFragment : Fragment(), FragmentResultListener {
         super.onViewCreated(view, savedInstanceState)
         childFragmentManager.setFragmentResultListener(DIALOG_DATE, viewLifecycleOwner, this)
         observeData()
+        setOnBackPressed()
+    }
+
+    private fun loadCrimeFromArguments() {
+        val crimeId: Int = arguments?.getSerializable(ARG_CRIME_ID) as Int
+        crimeDetailViewModel.loadCrime(crimeId)
+    }
+
+    private fun setOnBackPressed() {
```

Comment:
As a personal preference I would name this to `showDialogOnBackPressed`. I will approve if you prefer to leave it this way